crates/chia-datalayer/src/lib.rs

```diff
@@ -1,42 +1,155 @@
-use std::collections::HashMap;
+// use std::collections::HashMap;
 
-type Index = usize;
-type Key = Vec<u8>;
+type TreeIndex = u32;
+// type Key = Vec<u8>;
+type Hash = [u8; 32];
+type KVId = Hash;
+
+#[derive(Debug, Hash, Eq, PartialEq)]
 
 pub enum NodeType {
```

Comment:
This should also derive `Clone, Copy, Hash` IMO

Reply:
I have been adding the derives as I find a need.  I expect once I'm more familiar I will pre-emptively add some as well.

Reply:
Doesn't matter if nobody's using it I guess but a good habit to get into